src/test/scala/mesosphere/marathon/state/GroupManagerTest.scala

```diff
@@ -87,7 +87,80 @@ class GroupManagerTest extends MarathonActorSupport with MockitoSugar with Match
     ))
     val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
     update.transitiveApps.filter(_.hasDynamicPort) should be ('empty)
-    update.transitiveApps.flatMap(_.portNumbers.filter(x => x >= 10 && x <= 20)) should have size 2
+    update.transitiveApps.flatMap(_.hostPorts.flatten.filter(x => x >= 10 && x <= 20)) should have size 2
+  }
+
+  test("Assign dynamic service ports specified in multiple containers") {
+    import Container.Docker
+    import Docker.PortMapping
+    import org.apache.mesos.Protos.ContainerInfo.DockerInfo.Network
+    val c1 = Some(Container(
+      docker = Some(Docker(
+        image = "busybox",
+        network = Some(Network.USER),
+        portMappings = Some(Seq(
+          PortMapping(containerPort = 8080)
+        ))
+      ))
+    ))
+    val c2 = Some(Container(
+      docker = Some(Docker(
+        image = "busybox",
+        network = Some(Network.USER),
+        portMappings = Some(Seq(
+          PortMapping(containerPort = 8081)
+        ))
+      ))
+    ))
+    val group = Group(PathId.empty, Set(
+      AppDefinition("/app1".toPath, portDefinitions = Seq(), container = c1),
+      AppDefinition("/app2".toPath, portDefinitions = Seq(), container = c2)
+    ))
+    val update = manager(minServicePort = 10, maxServicePort = 20).assignDynamicServicePorts(Group.empty, group)
```

Comment:
I think that if this test used the range [10-12), then it would be more likely to fail if the service ports becomes random (with duplicates).